client/my-sites/upgrades/domain-management/list/index.jsx

```diff
@@ -82,16 +209,79 @@ const List = React.createClass( {
 		);
 	},
 
+	setPrimaryDomain: function( domainName ) {
+		return new Promise( ( resolve, reject ) => {
+			setPrimaryDomain( this.props.selectedSite.ID, domainName, ( error, data ) => {
+				if ( ! error && data && data.success ) {
+					page.redirect( paths.domainManagementList( this.props.selectedSite.slug ) );
+					resolve();
+				} else {
+					reject( error );
+				}
+			} )
+		} );
+	},
+
+	handleUpdatePrimaryDomain( index, domain ) {
+		if ( this.state.settingPrimaryDomain ) {
+			return;
+		}
+
+		this.recordEvent( 'changePrimary', domain );
+		const currentPrimary = find( this.props.domains.list, { isPrimary: true } ).name;
+
+		if ( domain.name === currentPrimary ) {
+			// user clicked the current primary domain
+			this.setState( {
+				primaryDomainModeEnabled: false
```

Comment:
Aaargh, missed this one - this should be `changePrimaryDomainModeEnabled` I think. Especially since `primaryDomainModeEnabled` is never referenced again ;)